airbyte-integrations/connectors/source-confluence/source_confluence/source.py

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from base64 import b64encode
+from typing import Any, Dict, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.requests_native_auth import TokenAuthenticator
+from airbyte_cdk.sources.utils.transform import TransformConfig, TypeTransformer
+
+
+# Basic full refresh stream
+class ConfluenceStream(HttpStream, ABC):
+    url_base = "https://{}/wiki/rest/api/"
+    primary_key = "id"
+    limit = 50
+    start = 0
+    expand = []
+    transformer: TypeTransformer = TypeTransformer(TransformConfig.DefaultSchemaNormalization)
+
+    def __init__(self, config: Dict):
+        super().__init__(authenticator=config["authenticator"])
+        self.config = config
+        self.url_base = self.url_base.format(config["domain_name"])
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        json_response = response.json()
+        links = json_response.get("_links")
+        next_link = links.get("next")
+        if next_link:
+            self.start += self.limit
+            return {"start": self.start}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {"limit": self.limit, "expand": ",".join(self.expand)}
+        if next_page_token:
+            params.update({"start": next_page_token["start"]})
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        json_response = response.json()
+        records = json_response.get("results", [])
+        yield from records
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return self.api_name
+
+
+class BaseContentStream(ConfluenceStream, ABC):
+    api_name = "content"
+    expand = [
+        "history",
+        "history.lastUpdated",
+        "history.previousVersion",
+        "history.contributors",
+        "restrictions.read.restrictions.user",
+        "version",
+        "descendants.comment",
+    ]
+    limit = 25
+    content_type = None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        params.update({"type": self.content_type})
+        return params
+
+
+class Pages(BaseContentStream):
+    content_type = "page"
+
```

Comment:
Also do you mind adding docs url here for every stream ?

Reply:
^ @tuanchris

Reply:
I have added the documentation strings